File: KutuphaneAPI/Entities/RequestFeatures/BookRequestParameters.cs
namespace Entities.RequestFeatures
{
    public class BookRequestParameters : RequestParameters
    {
        public int? AuthorId { get; set; }
        public int? CategoryId { get; set; }
        public bool? IsAvailable { get; set; }
        public String? Fields { get; set; }
        public bool? IsPopular { get; set; }
        public ICollection<int>? CategoryIds { get; set; }
        public ICollection<int>? TagIds { get; set; }
    }
}

File: KutuphaneWeb/library-automation-web/src/types/bookRequestParameters.ts
export default interface BookRequestParameters extends RequestParameters {
    fields?: string;
    authorId?: number;
    categoryId?: number;
    isPopular?: boolean;
    isAvailable?: boolean;
    tagIds?: number[];
    categoryIds?: number[];
}

export interface RequestParameters {
    searchTerm?: string;
    orderBy?: string;
    pageNumber?: number;
    pageSize?: number;
}